cookbooks/aws-parallelcluster-slurm/files/default/head_node_slurm/slurm/pcluster_slurm_config_generator.py

```diff
@@ -17,18 +17,293 @@
 import re
 from os import makedirs, path
 from socket import gethostname
-from typing import Tuple
+from typing import Dict, List, Tuple
 from urllib.parse import ParseResult, urlparse
 
 import requests
 import yaml
 from jinja2 import FileSystemLoader
 from jinja2.sandbox import SandboxedEnvironment
 
+CONFIG_HEADER = "# This file is automatically generated by pcluster"
 METADATA_REQUEST_TIMEOUT = 60
 log = logging.getLogger()
 instance_types_data = {}
 
+# Note we have a mismatch between python path configuration for the unit tests and the integration tests that
+# prevents us to move this code in dedicated files. We will address this issue in a different task.
+# As of now we will keep these changes in the same file
+
+
+# These functions should be moved into
+# config_utils.py
+def _get_instance_types(compute_resource_config) -> List[str]:
+    """Return the InstanceTypes defined in the ComputeResource always as list."""
+    if compute_resource_config.get("Instances"):
+        return [instance.get("InstanceType") for instance in compute_resource_config["Instances"]]
+    else:
+        return [compute_resource_config["InstanceType"]]
+
+
+def _get_efa_settings(compute_resource_config) -> Tuple[bool, bool]:
+    """Return a Tuple with EFA flags."""
+    if "Efa" in compute_resource_config:
+        efa = compute_resource_config["Efa"]
+        return efa["Enabled"], efa["GdrSupport"]
+    else:
+        return False, False
```

Comment:
Let's create an internal NTH task to add a unit test to cover the `else:` case. There are other occurrences where this is the case apparently (see below).

Reply:
I added this case in the following PR to be sure it was covered.